exporter/cortexexporter/cortex.go

```diff
@@ -0,0 +1,354 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cortexexporter
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/prompb"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/consumer/pdatautil"
+	"go.opentelemetry.io/collector/internal/data"
+	otlp "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/metrics/v1"
+)
+
+// TODO: get default labels such as job or instance from Resource
+
+// cortexExporter converts OTLP metrics to Cortex TimeSeries and sends them to a remote endpoint
+type cortexExporter struct {
+	namespace string
+	endpoint  string
+	client    *http.Client
+	headers   map[string]string
+	wg        *sync.WaitGroup
+	closeChan chan struct{}
+}
+
+// newCortexExporter initializes a new cortexExporter instance and sets fields accordingly.
+// client parameter cannot be nil.
+func newCortexExporter(ns string, ep string, client *http.Client) (*cortexExporter, error) {
+
+	if client == nil {
```

Comment:
should also check ep is valid url. perhaps require ep to be a URL rather than string

Reply:
What does `ep` stand for? Endpoint? That feels uncommon to me? May be use a clearer name